src/private/widgets/TabWidgetWidget.cpp

```diff
@@ -201,3 +205,34 @@ void TabWidgetWidget::updateMargins()
     m_cornerWidgetLayout->setContentsMargins(QMargins(0, 0, 2, 0) * factor);
     m_cornerWidgetLayout->setSpacing(int(2 * factor));
 }
+
+void TabWidgetWidget::contextMenuRequested(QPoint pos)
+{
+    auto *tabBar = QTabWidget::tabBar();
+    auto count = tabBar->count();
+    // We dont want context menu if there is only one tab
+    if (count <= 1) {
+        return;
+    }
+
+    // Click on a tab => No menu
+    if (tabBar->tabAt(pos) >= 0) {
+        return;
+    }
+
+    // Right click is allowed only on the tabs area
+    auto tabAreaRect = tabBar->rect();
+    tabAreaRect.setWidth(this->width());
+    if (!tabAreaRect.contains(pos)) {
+        return;
+    }
+
+    QMenu menu(this);
+
+    for (int i = 0; i < count; ++i) {
+        menu.addAction(tabText(i), this, [this, i] {
```

Comment:
disable the action for the current tab, as pressing it won't do anything

Reply:
Done